dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java

```diff
@@ -514,7 +514,9 @@ private static synchronized void registerDeadlockDetectionEvent(URL bootstrapUrl
   /** Enable JMX based system access provider once it is safe to touch JMX */
   private static synchronized void initializeJmxSystemAccessProvider(
       final ClassLoader classLoader) {
-    log.debug("Initializing JMX system access provider for " + classLoader.toString());
+    if (log.isDebugEnabled()) {
+      log.debug("Initializing JMX system access provider for " + classLoader.toString());
+    }
```

Comment:
Would there be any value to switching to the following like is done in the other file?

```suggestion
    log.debug("Initializing JMX system access provider for {}", classLoader);
```

Reply:
It would make the code feel more like the other code which could be a benefit. I would still keep it wrapped with the if statement to save the call before the code get JIT compiled. In theory we should wrap all of the the logging calls to remove the call from interpreted calls when Debug is not enabled.

Reply:
I will just leave it as is